### PR TITLE
Add project overrides to gke services

### PR DESCRIPTION
**Pull request by venkykuberan**
If this PR is for Terraform, I acknowledge that I have:

- [X ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement

container  : added project override support to `google_container_cluster` and `google_container_nodepool`
servicemanagement : added project override support to `google_project_service`

```
Fixes : https://github.com/hashicorp/terraform-provider-google/issues/6914

### CODE REVIEW

File: third_party/terraform/resources/resource_container_cluster.go.erb [hunk 1]
```diff
@@ -1407,7 +1407,11 @@ func resourceContainerClusterCreate(d *schema.ResourceData, meta interface{}) er
 	parent := fmt.Sprintf("projects/%s/locations/%s", project, location)
```
[the remark this answers is not in the document]
Comment (rileykarson) [reply third_party/terraform/resources/resource_container_cluster.go.erb:1407]: Whoops, you're right- I expected the change to look more like the node pool one.

File: third_party/terraform/tests/resource_google_project_service_test.go [hunk 2]
```diff
@@ -187,6 +187,9 @@ func testAccCheckProjectService(t *testing.T, services []string, pid string, exp
 
 func testAccProjectService_basic(services []string, pid, name, org string) string {
 	return fmt.Sprintf(`
+provider "google" {
+	user_project_override = true
```
Comment (rileykarson) [inline third_party/terraform/tests/resource_google_project_service_test.go:191]:
```suggestion
  user_project_override = true
```

File: third_party/terraform/resources/resource_container_node_pool.go.erb [hunk 3]
```diff
@@ -658,6 +699,10 @@ func flattenNodePool(d *schema.ResourceData, config *Config, np *containerBeta.N
 
 func nodePoolUpdate(d *schema.ResourceData, meta interface{}, nodePoolInfo *NodePoolInformation, prefix string, timeout time.Duration) error {
 	config := meta.(*Config)
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
```
[the remark this answers is not in the document]
Comment (rileykarson) [reply third_party/terraform/resources/resource_container_node_pool.go.erb:679]: I believe these calls still need to be updated to use `nodePoolInfo.project` as well, right?

**Comment by danawillow** [comment]
FYI this changelog note did not pass the changelog generator. I'm not sure whether the issue is the extra empty lines or having two messages in one changelog note, but just an FYI.

**Comment by danawillow** [comment]
(and by that I mean, it won't pass the changelog generator next week, which is when this will actually appear. I messed up the command this week but fixed it now)

**Pull request merged**